react/features/base/tracks/middleware.js

```diff
@@ -64,7 +72,19 @@ MiddlewareRegistry.register(store => next => action => {
         return result;
     }
     case TRACK_REMOVED: {
-        _removeNoDataFromSourceNotification(store, action.track);
+        const { track } = action;
+        const { jitsiTrack } = track;
+
+        if (jitsiTrack.videoType === 'desktop' && typeof APP === 'object') {
+            const participantId = _getJitsiTrackParticipantId(store, jitsiTrack);
```

Comment:
Maybe you can use `getTrackByJitsiTrack` to get the track from the store. This way you will have the videoType and the participantID from the track object in the store. WDYT?